### DISCUSSION TITLE
Setting numlock and capslock on Wayland using InputConfig

### DISCUSSION THREAD

**Discussion by otaj** [qtile/qtile#4005] [Ideas]
Hi, would it be possible to add support for setting numlock and capslock using `InputConfig`?

I started digging through the code and how is sway doing it and it seems it should be easily possible, however as I have zero experience developing qtile (actually started using it just very recently), I feel a bit overwhelmed by the code. I think I should be able to open a PR myself in some time, but I am honestly not sure, any guidance would be greatly appreciated.

Thanks!

**Comment by otaj** [comment]
Btw, I know there was a recent discussion asking if it's possible to start it using `numlockx`. I know that won't work on Wayland, since it has to be done on compositor side and this is more of a feature request.

**Comment by m-col** [comment]
Hey, apologies for missing this. I'm unsure exactly what would be required for implementing support for those. I think your best bet would be trying to understand from the source and github issues for sway, or perhaps a more simple compositor such as dwl. Most of the wlroots and xkbcommon code we use follows much of the same structure/API as their C equivalents, so it would then be a matter of translating it into the Python version and adding that to qtile.
